dsc_lib/src/dscresources/resource_manifest.rs

```diff
@@ -80,11 +80,23 @@ pub enum ManifestSchemaUri {
     VSCode2023_08,
 }
 
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize, JsonSchema)]
+#[serde(untagged)]
+pub enum ArgKind {
+    /// The argument is a string.
+    String(String),
+    /// The argument accepts the JSON input object.
+    Json{
+        /// The argument that accepts the JSON input object.
+        #[serde(rename = "jsonInputArg")]
+        json_input_arg: String,
+        /// Indicates if argument is mandatory which will pass an empty string if no JSON input is provided.  Default is false.
+        mandatory: Option<bool>,
+    }
+}
+
 #[derive(Debug, Clone, PartialEq, Deserialize, Serialize, JsonSchema)]
 pub enum InputKind {
-    /// The input replaces arguments with this token in the command.
-    #[serde(rename = "arg")]
-    Arg(String),
```

Comment:
Rather than deleting this option entirely, I think it makes sense to allow resource authors to explicitly define the input kind as `arg` to preclude being sent the property bag as environment variables or over stdin and accounting for double processing.

If we do this, we could also update the JSON Schema (and code) to infer the input kind from the manifest - if the args array includes an object and `input` isn't defined, `input` is `arg`. If the args array is only made of strings and `input` isn't defined, `input` is `stdin`. The `input` is only ever `env` if it's explicitly defined.

We could then also use JSON schema to validate the `args` array - if it's explicitly defined as `env` or `stdin`, is must _not_ include an object, only strings. If it's not explicitly defined or is defined as `arg`, it may only include one object in the array.

We could use this to provide better author-time feedback and validation for the resource manifest and direct authors towards best practices.

Reply:
Quick json schema example (a little pseudo code, bc it's a little complex to write initially but functional/useful):

```yaml
# set property in manifest
input:
  type: string
  enum: [arg, env, stdin]
args:
  type: array
  contains: { type: object } # check if array includes objects
  minContains: 0 # allow string-only
  maxContains: 1 # allow json arg only once
  items:
    anyOf:
      - type: string
      - type: object # validate object structure
        required: [jsonInputArg]
        properties:
          jsonInputArg:
            type: string
          mandatory:
            type: boolean
            default: false
# Helper for validation/authoring
allOf:
  - if: # args items are only strings
    then:
      properties:
        input: { default: stdin }
  - if: # args items includes one object
    then:
      properties:
        input: { default: arg }
  - if: # input is arg (default or explicit)
      properties: { input: { const: arg } }
    then: # require object arg exactly once
      properties:
        args: { minContains: 1 }
    else: # forbid object arg entirely
      properties:
        args: { maxContains: 0 }
```

Reply:
The resource author has to opt into any acceptance of the JSON whether it's env, stdin, or args so there isn't a case where it's accidentally sent.  However, there might be a case where the resource might want the simplified reception of the JSON as env, but also keep a raw copy as an arg to pass to something else.